var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -52,39 +65,225 @@ class Faiss(AutotoolsPackage, CudaPackage):
+            "gpu/test": [
+                "TestGpuIndexFlat",
+                "TestGpuIndexBinaryFlat",
+                "TestGpuIndexIVFFlat",
+                "TestGpuIndexIVFPQ",
+                "TestGpuMemoryException",
+                "TestGpuSelect",
+                "demo_ivfpq_indexing_gpu",
+            ]
+        }
+
+        return (ctargets, gtargets), (cinstalls, ginstalls)
+
+    def fetch_targets_v172(self):
+
+        ctargets = [
+            "demo_imi_flat",
+            "demo_imi_pq",
+            "demo_ivfpq_indexing",
+            "demo_nndescent",
+            "demo_sift1M",
+            "demo_weighted_kmeans",
+        ]
+
+        gtargets = [
+            "TestCodePacking",
+            "TestGpuDistance",
+            "TestGpuIndexBinaryFlat",
+            "TestGpuIndexFlat",
+            "TestGpuIndexIVFFlat",
+            "TestGpuIndexIVFPQ",
+            "TestGpuIndexIVFScalarQuantizer",
+            "TestGpuSelect",
+            "TestGpuMemoryException",
+            "demo_ivfpq_indexing_gpu",
+        ]
+
+        cinstalls = {"tests": ["faiss_test"], "demos": ctargets}
+        ginstalls = {"faiss/gpu/test": gtargets}
+
+        return (ctargets, gtargets), (cinstalls, ginstalls)
+
+    # --------------------------------------------------------------------------
+    @when("@1.7.2:")
+    def cmake_args(self):
+
+        variants2cmake = {
+            "cuda": "FAISS_ENABLE_GPU",
+            "python": "FAISS_ENABLE_PYTHON",
+            "tests": "BUILD_TESTING",
+            "shlib": "BUILD_SHARED_LIBS",
+        }
+
+        cmake_args = []
+        for v, c in variants2cmake.items():
+            if "+" + v in self.spec:
+                cmake_args.append("-D" + c + "=ON")
+            else:
+                cmake_args.append("-D" + c + "=OFF")
+
+        return cmake_args
+
+    # --------------------------------------------------------------------------
+    def build(self, spec, prefix):
+
+        with working_dir(self.build_directory):
+
+            # make the faiss build
+            make("faiss")
+
+            # build python bindings
+            if "+python" in self.spec:
+                make("swigfaiss")
+
+            if "+tests" not in self.spec:
+                return
+
+            # build tests
+            (ctargets, gtargets), (_, _) = self.fetch_targets_v172()
+
+            # CPU tests
+            make("gtest")
+            for t in ctargets:
+                make(t)
+
+            # GPU tests
+            if "+cuda" in self.spec:
+                for t in gtargets:
+                    make(t)
+
+    # --------------------------------------------------------------------------
+    def install(self, spec, prefix):
+
+        with working_dir(self.build_directory):
+
+            make("install")
+            if "+python" in self.spec:
+                with working_dir("faiss/python"):
+                    args = std_pip_args + ["--prefix=" + prefix, "."]
+                    pip(*args)
```

Comment:
This can be a function on its own:
```python
@run_after('install')
def install_python_package(self):
    ...
```
at which point you don't need to override `install`.